python/pyspark/sql/tests/test_connect_compatibility.py

```diff
@@ -249,7 +267,22 @@ def test_spark_session_compatibility(self):
         """Test SparkSession compatibility between classic and connect."""
         expected_missing_connect_properties = {"sparkContext"}
         expected_missing_classic_properties = {"is_stopped", "session_id"}
-        expected_missing_connect_methods = {"newSession"}
+        expected_missing_connect_methods = {
+            "addArtifact",
+            "addArtifacts",
+            "addTag",
+            "clearProgressHandlers",
+            "clearTags",
+            "copyFromLocalToFs",
+            "getTags",
+            "interruptAll",
+            "interruptOperation",
+            "interruptTag",
+            "newSession",
+            "registerProgressHandler",
+            "removeProgressHandler",
+            "removeTag",
+        }
```

Comment:
Now the compatibility check respect the `ONLY_SUPPORTED_WITH_SPARK_CONNECT`.

cc @HyukjinKwon @hvanhovell FYI